File: src/main/java/org/sylfra/idea/plugins/remotesynchronizer/ProjectStartupActivity.java
package org.sylfra.idea.plugins.remotesynchronizer;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.startup.StartupActivity;
import org.jetbrains.annotations.NotNull;

/**
 *
 * <p>
 * date: 2023/2/19 18:57
 *
 * @author Zhengyu Hu
 */
public class ProjectStartupActivity implements StartupActivity {

    @Override
    public void runActivity(@NotNull Project project) {
        project.getService(FileSyncPlugin.class).projectOpened();
    }
}

File: src/main/kotlin/org/imyuyu/idea/plugins/filesync/ProjectStartupActivity.kt
package org.imyuyu.idea.plugins.filesync

import com.intellij.openapi.project.Project
import com.intellij.openapi.startup.ProjectActivity
import com.intellij.openapi.startup.StartupActivity

/**
 *
 * Startup Activity
 *
 * date: 2023/2/19 18:57
 *
 * @author Zhengyu Hu
 */
class ProjectStartupActivity : ProjectActivity {

    override suspend fun execute(project: Project) {
        FileSyncPlugin.getInstance(project).projectOpened()
    }
}